mediapipe/tasks/c/components/containers/matrix_converter.cc

```diff
@@ -0,0 +1,45 @@
+/* Copyright 2023 The MediaPipe Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "mediapipe/tasks/c/components/containers/matrix_converter.h"
+
+#include <cstdlib>
+
+#include "Eigen/Core"
+#include "mediapipe/tasks/c/components/containers/matrix.h"
+
+namespace mediapipe::tasks::c::components::containers {
+
+void CppConvertToMatrix(const Eigen::MatrixXf& in, ::Matrix* out) {
+  out->rows = in.rows();
+  out->cols = in.cols();
+  out->data = new float[out->rows * out->cols];
+
+  // Copy data from Eigen matrix to C matrix in column-major order
+  for (int col = 0; col < out->cols; ++col) {
```

Comment:
Can we avoid the one by one copy here by using https://eigen.tuxfamily.org/dox/classEigen_1_1PlainObjectBase.html#ac54123f62de4c46a9107ff53890b6116 ? Note that the order of elements does not really matter here as long as we document it.

Reply:
@schmidt-sebastian I've documented it now and simplified the copying process.